lib/virt_autotest/utils.pm

```diff
@@ -105,10 +105,23 @@ sub is_xen_host {
     return get_var("XEN") || check_var("SYSTEM_ROLE", "xen") || check_var("HOST_HYPERVISOR", "xen") || check_var("REGRESSION", "xen-hypervisor");
 }
 
+#check kernels by grep keywords from journals
+#welcome everybody to extend this function
+sub check_kernel {
+    my $machine //= 'localhost';
+
+    my @oops_keywords = ('Started Process Core Dump', 'Call Trace');
+    foreach my $keyword (@oops_keywords) {
+        my $cmd = "journalctl | grep '$keyword'";
+        $cmd = "ssh root\@$machine " . "\"$cmd\"" if $machine ne 'localhost';
+        record_soft_failure("$keyword in $machine fg. bsc#1204727") if script_run($cmd) == 0;
```

Comment:
Hi Julie, do I understand correctly that the current code logic is that once we find above oops_keywords in journal log, the test always record_soft_failure bsc#1204727? IMHO, the oops_keywords are general key words for coredump or call trace, but not necessarily this specific bug? And once such keywords found, the tester should check in details to see if NEW BUG needs to be opened?

Reply:
because of the PR CI check polity, a record_soft_failure must be accociate with a poo or bsc number. I have to put it here, or use a face 'bsc#5555'? which one do you prefer?

Reply:
Record_soft_fail is easy to be ignored directly. Maybe we should make it die first with warning messages like  coredump or call trace found? Then if after tester's analysis ,
-  no existing bug for that specific issue, open new bug in bugzilla first
- if the bug won't block further test, add specific record_soft_failure with bsc#num, to let test continue
- if the bug is blocking for further tests, then let the tests fail just as it is, no more code change
HDYT?

Reply:
Record_soft_fail is easy to be ignored directly. Maybe we should make it die first with warning messages like  coredump or call trace XX found? Then if after tester's analysis ,
-  no existing bug for that specific issue, open new bug in bugzilla first
- if the bug won't block further test, add specific record_soft_failure with bsc#num, to let test continue
- if the bug is blocking for further tests, then let the tests fail just as it is, no more code change
HDYT?

Reply:
I prefer a soft failure, because:

- in my case, either a coredump or a call trace does not block our consequent tests possiblely.
- if 'die' was used, how do we know whether it would block further test?

To raise the attention of the coredupm/call trace warnings in our test, we can get all members involved this PR.

Reply:
> I prefer a soft failure, because:
> 
> * in my case, either a coredump or a call trace does not block our consequent tests possiblely.
> * if 'die' was used, how do we know whether it would block further test?
> 

Good question. I asked myself also this when giving the suggestion. This is the main drawback. It would need separate test to know whether it is soft fail or hard fail. This is not good indeed. However, I guess it matches why record_soft_fail would need bsc# (similar reasons)?

It is hard to choose indeed. Neither of them are good enough :(

If we choose soft-fail, I am afraid adding fake bsc# or poo# would obey the rule essentially, although the CI check passes?


> To raise the attention of the coredupm/call trace warnings in our test, we can get all members involved this PR.

Yes, this will be needed if soft-fail is chosen. Besides, if choosing soft-fail, we also need to figure out some way to make coredupm/call trace  soft-fail more obvious/differentiated from other real known soft-fail. Otherwise, practically, it may easy to loose attention, or to require checking all soft-fail which is time-consuming?